aspnetcore/host-and-deploy/windows-service.md

```diff
@@ -2,59 +2,112 @@
+
 ## Convert a project into a Windows Service
 
-The following minimum changes are required to set up an existing ASP.NET Core project to run as a service:
+Make the following changes to an existing ASP.NET Core project to run the app as a service:
 
 1. In the project file:
 
-   * Confirm the presence of a Windows [Runtime Identifier (RID)](/dotnet/core/rid-catalog) or add it to the `<PropertyGroup>` that contains the target framework:
+   * **Framework-dependent Deployment (FDD)** &ndash; Add a Windows [Runtime Identifier (RID)](/dotnet/core/rid-catalog) to the `<PropertyGroup>` that contains the target framework. Add the `<SelfContained>` property set to `false`. Disable the creation of a *web.config* file by adding the `<IsTransformWebConfigDisabled>` property set to `true`.
+
+     ```xml
+     <PropertyGroup>
+       <TargetFramework>netcoreapp2.2</TargetFramework>
+       <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
+       <SelfContained>false</SelfContained>
+       <IsTransformWebConfigDisabled>true</IsTransformWebConfigDisabled>
+     </PropertyGroup>
+     ```
 
-      ```xml
-      <PropertyGroup>
-        <TargetFramework>netcoreapp2.2</TargetFramework>
-        <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
-      </PropertyGroup>
-      ```
+     **Self-contained Deployment (SCD)** &ndash; Confirm the presence of a Windows [Runtime Identifier (RID)](/dotnet/core/rid-catalog) or add a RID to the `<PropertyGroup>` that contains the target framework. Disable the creation of a *web.config* file by adding the `<IsTransformWebConfigDisabled>` property set to `true`.
 
-      To publish for multiple RIDs:
+     ```xml
+     <PropertyGroup>
+       <TargetFramework>netcoreapp2.2</TargetFramework>
+       <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
+       <IsTransformWebConfigDisabled>true</IsTransformWebConfigDisabled>
+     </PropertyGroup>
+     ```
 
-      * Provide the RIDs in a semicolon-delimited list.
-      * Use the property name `<RuntimeIdentifiers>` (plural).
+     To publish for multiple RIDs:
 
-      For more information, see [.NET Core RID Catalog](/dotnet/core/rid-catalog).
+     * Provide the RIDs in a semicolon-delimited list.
+     * Use the property name `<RuntimeIdentifiers>` (plural).
+
+     For more information, see [.NET Core RID Catalog](/dotnet/core/rid-catalog).
 
    * Add a package reference for [Microsoft.AspNetCore.Hosting.WindowsServices](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.WindowsServices).
 
+   * If the app uses a custom <xref:Microsoft.AspNetCore.Hosting.WindowsServices.WebHostService>, set a `Component` subtype on the file that contains the `WebHostService` implementation. The sample app includes a custom `WebHostService` in its *CustomWebHostService.cs* file that handles app starting and stopping events:
+
+     ```xml
+     <ItemGroup>
+       <Compile Update="CustomWebHostService.cs">
+         <SubType>Component</SubType>
```

Comment:
Just curious..what does this do? I don't think it is actually required. Is this for some Visual Studio feature?

Reply:
I inherited that from the prior authors of the topic+sample ... perhaps @tdykstra knows why it was there (or perhaps Sourabh). I assumed it was required, and I didn't run a test without it.

Reply:
@guardrex If I did know, I've long since forgotten. 😄

Reply:
@tdykstra I know the feeling my friend. lol .... that happens to me more often than I'd like to admit. Thank the :pray: *server gods* :pray: for GH search! lol

@shirhatti ... do you know why the Windows Service sample has this in its project file. It goes way, *way* back to when the topic was first written I think ...

```xml
<ItemGroup>
  <Compile Update="CustomWebHostService.cs">
    <SubType>Component</SubType>
  </Compile>
</ItemGroup>
```

Reply:
My hunch is that this isn't necessary anymore. I'd favor removing it since we don't know what it does...but it also doesn't harm anything by being present since it's "just metadata" which doesn't affect the way the app runs.

Reply:
I'll run a quick test on it. If the bits in `CustomWebHostService` work ok, I'll kill it off on the next commit.

Reply:
No idea 😕

Reply:
The conclusion is that *I'm in idiot.* ... but we knew that already. The events appear **_only if_** I use the `Default` log level setting in the settings file. I ran into trouble with other approaches. It's fine that this sample uses the approach that I'll have here. When Tom gets to updating our docs on *the right way to do it*, I look forward to seeing what he suggests in the topic updates. If you would like to try anything different, shoot me a plan of action after the next commit goes in.